scripts/hash-exec-copy.sh

```diff
@@ -1,13 +1,34 @@
 #!/usr/bin/env bash
+
 set -e
 
-[[ "$1" == https://raw.githubusercontent.com/makerdao/community/*/governance/votes/*.md ]] || { echo "Please provide the correct exec copy link to hash (e.g. url=https://raw.githubusercontent.com/makerdao/community/<commit>/governance/votes/<file name>.md)"; exit 1; }
+if [[ -z "$1" ]]; then
+echo "Error: Please provide a date in YYYY-MM-DD format as an argument."
+exit 1
+fi
+
+# Check input date for correct format
+if ! [[ $1 =~ ^[0-9]{4}-[0-9]{2}-[0-9]{2}$ ]] ; then
+    echo 'Invalid date format. Please use yyyy-mm-dd'
+    exit 1
+fi
+
+# Format input date into GitHub date URL format
+DATE=$(date -d "$1" +"%B %d, %Y" | sed -e 's/ /%20/g' -e 's/,/%2C/g')
 
-if [[ -x "$(command -v wget)" ]]; then
-    cast keccak -- "$(wget "$1" -q -O - 2>/dev/null)"
-elif [[ -x "$(command -v curl)" ]]; then
-    cast keccak -- "$(curl "$1" -o - 2>/dev/null)"
+if [[ -x "$(command -v wget)" ]] && [[ -x "$(command -v curl)" ]]; then
+    # Get latest change git commit hash for target exec copy
+    commit_hash=$(curl -s "https://api.github.com/repos/makerdao/community/commits?path=governance/votes/Executive%20vote%20-%20$DATE.md&per_page=1" | grep -o -E '"sha": "[^"]+"' | head -n 1 | awk '{print $2}' | sed 's/"//g')
+    [[ -z "$commit_hash" ]] && { echo "Error: Executive vote not found"; exit 1; }
+    # Hash target exec copy
+    exec_hash=$(cast keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/"$commit_hash"/governance/votes/Executive%20vote%20-%20"$DATE".md -q -O - 2>/dev/null)")
 else
-    echo "Please install either wget or curl";
+    echo "Please ensure you have curl and wget installed";
```

Comment:
can't we use just one? why require that they are both installed?

Reply:
yeah, we could, I've added both for portability but we could stick with `curl` I guess.

Reply:
done in 0bd2452